windows build failure linking node.dll: v8.lib was created with an older compiler than other objects

* Electron Version: master (e79751c8d450249b19c764f79c9d741f2662b074)
* Operating System (Platform and Version): win7 sp1 x64

**Expected Behavior**
should build

**Actual behavior**
```
[9/205] LINK_EMBED(DLL) node.dll
FAILED: C:\Python27\python.exe gyp-win-tool link-with-manifests environment.x86 True node.dll "C:\Python27\python.exe gyp-win-tool l
ink-wrapper environment.x86 False link.exe /nologo /IMPLIB:node.dll.lib /DLL /OUT:node.dll @node.dll.rsp" 2 mt.exe rc.exe "obj\vendo
r\node\node_lib.node.dll.intermediate.manifest" obj\vendor\node\node_lib.node.dll.generated.manifest
fatal error C1047: The object or library file 'C:\src\electron\vendor\download\libchromiumcontent\static_library\v8.lib'
was created with an older compiler than other objects; rebuild old objects and libraries
LINK : fatal error LNK1257: code generation failed
Traceback (most recent call last):
  File "gyp-win-tool", line 323, in <module>
    sys.exit(main(sys.argv[1:]))
  File "gyp-win-tool", line 29, in main
    exit_code = executor.Dispatch(args)
  File "gyp-win-tool", line 71, in Dispatch
    return getattr(self, method)(*args[1:])
  File "gyp-win-tool", line 179, in ExecLinkWithManifests
    subprocess.check_call(ldcmd + add_to_ld)
  File "C:\Python27\lib\subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'C:\Python27\python.exe gyp-win-tool link-wrapper environment.x86 False link.exe /nologo /IMP
LIB:node.dll.lib /DLL /OUT:node.dll @node.dll.rsp node.dll.manifest.res' returned non-zero exit status 1257
[9/205] CXX obj\atom\browser\electron_lib.native_window_views.obj
ninja: build stopped: subcommand failed.
```

**To Reproduce**

do a clean windows 32bit build with VS2017 15.7.1

**Additional Information**

https://docs.microsoft.com/en-us/cpp/error-messages/compiler-errors-1/fatal-error-c1047

> C1047 is caused when object files or libraries built with /LTCG are linked together, but where those object files or libraries are built with different versions of the Visual C++ toolset. 

looks like /LTCG is the culprit

https://docs.microsoft.com/en-us/cpp/build/reference/ltcg-link-time-code-generation
https://docs.microsoft.com/en-us/cpp/build/reference/gl-whole-program-optimization

different vs2017 minor versions can't be installed side by side, so requiring a specific version does not seem practical

--- a/docs/development/build-instructions-windows.md
+++ b/docs/development/build-instructions-windows.md
@@ -5,8 +5,8 @@ Follow the guidelines below for building Electron on Windows.
 ## Prerequisites
 
 * Windows 7 / Server 2008 R2 or higher
-* Visual Studio 2017 15.7.2 or higher - [download VS 2017 Community Edition for
-  free](https://www.visualstudio.com/vs/)
+* [Visual Studio 2017 15.4.5](https://web.archive.org/web/20180220131834/https://docs.microsoft.com/en-us/visualstudio/productinfo/installing-an-earlier-release-of-vs2017) - Electron 3-0-x specifically requires Visual Studion 15.4.5 to build and newer versions will not work.  You can [download VS 2017 15.4.5 Community Edition for
+  free](https://aka.ms/eam4b7).
 * [Python 2.7](http://www.python.org/download/releases/2.7/)
 * [Node.js](https://nodejs.org/download/)
 * [Git](http://git-scm.com)

